sympy/stats/rv_interface.py

```diff
@@ -483,6 +483,45 @@ def coskewness(X, Y, Z, condition=None, **kwargs):
          * std(Z, condition, **kwargs)
     return num/den
 
+def self_information(condition, base = 2, **kwargs):
+    """
+    Calculates the self-information of a particular outcome of an event.
+    It represents the degree of surprise associated with a certain outcome.
+    Mathematically it is defined as
+
+    .. math::
+        selfinformation(p, b)=-\log_b (p)
+
+    Parameters
+    ==========
+
+    condition : Expression containing RandomSymbols
+    base : Base with respect to which the self-information is calculated
+
+    Examples
+    ========
+    >>> from sympy.stats import self_information, Die
+    >>> from sympy import Eq
+    >>> X = Die('X', 6)
+    >>> self_information(X>3)
+    1
+    >>> self_information(Eq(X,4))
+    1 + log(3)/log(2)
+    >>> self_information(X<2, 10)
+    log(6)/log(10)
+
+    Returns
+    =======
+
+    self-information : The self-information of the probability of an event
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Information_content
+    """
+    p = P(condition)
```

Comment:
I think `probability` has few other parameters too. Though I am not very supportive of this new function. After all, it will just help in saving a few key strokes.
@Upabjojr @Smit-create Should we have this addition?

Reply:
You're right, I'll try implementing a function which actually adds value to the stats module. This did help me learn how to navigate through the module while writing a new function.